### PR TITLE
Add DNS support for dialling nodes

### PR DESCRIPTION
**Pull request by akshar**
# Description
While bringing up new bootnodes with dns conn string in genesis file, it failed to connect to peers. Right now only ipv4/6 is supported.

This PR adds regex to dial via dns conn string. Reusing the regex used [here](https://github.com/0xPolygon/polygon-edge/blob/develop/helper/flags/flags.go#L55)

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

### CODE REVIEW

**Review by 0xAleksaOpacic — APPROVED** [review]
Well done! Thank you for this PR

File: network/server.go [hunk 1]
```diff
@@ -759,14 +759,16 @@ func StringToAddrInfo(addr string) (*peer.AddrInfo, error) {
...
 	// This regex will match:
 	// /ip4/localhost/tcp/<port>
 	// /ip4/127.0.0.1/tcp/<port>
 	// /ip4/<any other loopback>/tcp/<port>
 	// /ip6/<any loopback>/tcp/<port>
+	// /dns/foobar.com/tcp/<port>
 	loopbackRegex = regexp.MustCompile(
-		`^\/ip4\/127(?:\.[0-9]+){0,2}\.[0-9]+\/tcp\/\d+$|^\/ip4\/localhost\/tcp\/\d+$|^\/ip6\/(?:0*\:)*?:?0*1\/tcp\/\d+$`,
+		//nolint:lll
+		`^\/ip4\/127(?:\.[0-9]+){0,2}\.[0-9]+\/tcp\/\d+$|^\/ip4\/localhost\/tcp\/\d+$|^\/ip6\/(?:0*\:)*?:?0*1\/tcp\/\d+$|^\/dns\/[^-|^/][A-Za-z0-9-]([^-|^/]?)+([\\-\\.]{1}[a-z0-9]+)*\\.[A-Za-z]{2,}(/?)$`,
```
Comment (zivkovicmilos) [inline network/server.go]: What do you think about extracting the last part of the regex you've reused from [here](https://github.com/0xPolygon/polygon-edge/blob/develop/helper/flags/flags.go#L55) into the `common` or `helper` packages, and just referencing it here?

We use it in multiple places now with the introduction of this PR, and future fixes should propagate to all usages.

**Review by zivkovicmilos — APPROVED** [review]
Looks great 💯

File: network/server.go (hunk 1 above)
Comment (akshar) [reply network/server.go]: 👍🏼 . have extracted the regex to a simple helper func.  
Resolved here https://github.com/0xPolygon/polygon-edge/pull/406/commits/63c2bd6d1cd5d8f7ab3ad0a52b84a1de8baa9a6b

File: helper/common/common.go [hunk 2]
```diff
@@ -93,3 +93,8 @@ func GetTerminationSignalCh() <-chan os.Signal {
 
 	return signalCh
 }
+
+// GetDNSRegex returns a regex string  to match against a valid dns/dns4/dns6 addr
+func GetDNSRegex() string {
+	return `^/?(dns)(4|6)?/[^-|^/][A-Za-z0-9-]([^-|^/]?)+([\\-\\.]{1}[a-z0-9]+)*\\.[A-Za-z]{2,}(/?)$`
+}
```
Comment (Kourin1996) [inline helper/common/common.go]: You might want to define constant value instead of function?

**Review by Kourin1996 — CHANGES_REQUESTED** [review]

File: helper/common/common.go (hunk 2 above)
Comment (Kourin1996) [reply helper/common/common.go]: For now, we don't need to put this under `helper`. Why don't you define in network/server.go?

Comment (akshar) [reply helper/common/common.go]: Resolved here: https://github.com/0xPolygon/polygon-edge/pull/406/commits/2ad796dbf3e4884a9b15f72e9fa9059d88744282

**Review by Kourin1996 — APPROVED** [review]

File: network/server.go (hunk 1 above)
Comment (akshar) [reply network/server.go]: as suggested my @Kourin1996. defining a constant for now. resolved here https://github.com/0xPolygon/polygon-edge/commit/2ad796dbf3e4884a9b15f72e9fa9059d88744282

**Comment by zivkovicmilos** [comment]
Hey @akshar,

Please update this branch with changes from the latest develop, so we can merge this out 🙏

**Comment by zivkovicmilos** [comment]
@akshar 

Additionally, please make sure to reuse the same DNS regex reference in places in your other PR:
https://github.com/0xPolygon/polygon-edge/pull/405

**Comment by akshar** [comment]
> 

Merged latest develop.

> @akshar
> 
> Additionally, please make sure to reuse the same DNS regex reference in places in your other PR: #405

Keeping this in network/server.go as suggested by @Kourin1996 . let me know if you want to move it to helper or reference the const by importing it in [previous PR](https://github.com/0xPolygon/polygon-edge/pull/405/files) cc: @zivkovicmilos

**Pull request merged**